autogpt_platform/frontend/src/components/CustomNode.tsx

```diff
@@ -543,113 +574,185 @@ export function CustomNode({ data, id, width, height }: NodeProps<CustomNode>) {
+
+            <div className="mt-2 flex items-start justify-end rounded-b-xl pr-2">
+              <div className="flex-none">
+                {data.outputSchema &&
+                  generateOutputHandles(data.outputSchema, data.uiType)}
               </div>
-            </>
-          ) : (
-            <span>No outputs yet</span>
-          )}
-        </div>
-      )}
-      {data.uiType !== BlockUIType.NOTE && (
-        <div className="mt-2.5 flex items-center pb-4 pl-4">
-          <Switch checked={isOutputOpen} onCheckedChange={toggleOutput} />
-          <span className="m-1 mr-4">Output</span>
-          {hasAdvancedFields && (
-            <>
-              <Switch onCheckedChange={toggleAdvancedSettings} />
-              <span className="m-1">Advanced</span>
-            </>
-          )}
-          {data.status && (
-            <Badge
-              variant="outline"
-              data-id={`badge-${id}-${data.status}`}
-              className={cn(data.status.toLowerCase(), "ml-auto mr-5")}
+            </div>
+          </>
+        ) : (
+          <></>
+        )}
+      </div>
+      {/* End Body */}
+      {/* Footer */}
+      <div className="flex rounded-b-xl">
+        {/* Display Outputs  */}
+        {isOutputOpen && data.uiType !== BlockUIType.NOTE && (
+          <div
+            data-id="latest-output"
+            className={cn(
+              "nodrag w-full overflow-hidden break-words",
+              statusBackgroundClass,
+            )}
+          >
+            {(data.executionResults?.length ?? 0) > 0 ? (
+              <div className="mt-0 rounded-b-xl bg-gray-100">
+                <Separator.Root className="h-[2px] w-full bg-gray-200"></Separator.Root>
+                <NodeOutputs
+                  title="Latest Output"
+                  truncateLongData
+                  data={data.executionResults!.at(-1)?.data || {}}
+                />
+                <div className="flex justify-end">
+                  <Button variant="ghost" onClick={handleOutputClick}>
+                    View More
+                  </Button>
+                </div>
+              </div>
+            ) : (
+              <div className="mt-0 min-h-4 rounded-b-xl bg-white"></div>
+            )}
+            <div
+              className={cn(
+                "flex min-h-12 items-center justify-end",
+                statusBackgroundClass,
+              )}
             >
-              {data.status}
-            </Badge>
-          )}
-        </div>
-      )}
+              <Badge
+                variant="default"
+                className={cn(
+                  "mr-4 flex min-w-[114px] items-center justify-center rounded-3xl text-center text-xs font-semibold",
+                  hasConfigErrors || hasOutputError
+                    ? "border-red-600 bg-red-600 text-white"
+                    : {
+                        "border-green-600 bg-green-600 text-white":
```

Comment:
Should we group this mapping in the same place as the the coloring group above?

Color scheming has become a problem now, since we start using fixed color, we might need to put all these color schema in a single place.